### PR TITLE
add control loop to remove stale conditions

### PR DESCRIPTION
**Pull request by deads2k**
promote https://github.com/openshift/cluster-kube-scheduler-operator/pull/143 to library-go so auth operator can use it.

### CODE REVIEW

File: pkg/operator/staleconditions/remove_stale_conditions.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+func NewRemoveStaleConditions(
+	conditions []string,
+	operatorClient v1helpers.StaticPodOperatorClient,
+	eventRecorder events.Recorder,
+) *RemoveStaleConditions {
+	c := &RemoveStaleConditions{
+		conditions: conditions,
+
```
Comment (ravisantoshgudimetla) [inline pkg/operator/staleconditions/remove_stale_conditions.go]: unnecessary white space?

File: pkg/operator/staleconditions/remove_stale_conditions.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+func (c RemoveStaleConditions) sync() error {
+	removeStaleConditionsFn := func(status *operatorv1.StaticPodOperatorStatus) error {
+		for _, condition := range c.conditions {
+			v1helpers.RemoveOperatorCondition(&status.Conditions, condition)
+		}
+		return nil
+	}
+
```
Comment (ravisantoshgudimetla) [inline pkg/operator/staleconditions/remove_stale_conditions.go]: Same as above comment.

File: pkg/operator/staleconditions/remove_stale_conditions.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+const workQueueKey = "key"
+
+type RemoveStaleConditions struct {
+	conditions []string
+
+	operatorClient v1helpers.StaticPodOperatorClient
+	cachesToSync   []cache.InformerSynced
+
+	eventRecorder events.Recorder
```
Comment (enj) [inline pkg/operator/staleconditions/remove_stale_conditions.go]: `eventRecorder` is unused.

File: pkg/operator/staleconditions/remove_stale_conditions.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewRemoveStaleConditions(
+	conditions []string,
```
Comment (enj) [inline pkg/operator/staleconditions/remove_stale_conditions.go]: nit: I find `...string` more enjoyable to use

File: pkg/operator/staleconditions/remove_stale_conditions.go (hunk 3 above)
Comment (deads2k) [reply pkg/operator/staleconditions/remove_stale_conditions.go]: > `eventRecorder` is unused.

seems like that's a thing to be fixed at some later date.

File: pkg/operator/staleconditions/remove_stale_conditions.go (hunk 4 above)
Comment (deads2k) [reply pkg/operator/staleconditions/remove_stale_conditions.go]: > nit: I find `...string` more enjoyable to use

I don't agree for this one.  I think construction like this should be clear and explicit.  These tend to grow many params.

**Comment by enj** [comment]
/lgtm

Not worried about the nits.

**Pull request merged**